workflow/controller/controller.go

```diff
@@ -194,6 +205,45 @@ func (wfc *WorkflowController) Run(ctx context.Context, wfWorkers, podWorkers in
 	<-ctx.Done()
 }
 
+// Create and initialize the Synchronization Manager
+func (wfc *WorkflowController) createSynchronizationManager() error {
+	syncLimitConfig := func(lockName string) (int, error) {
+		items := strings.Split(lockName, "/")
+		if len(items) < 4 {
+			return 0, argoErr.New(argoErr.CodeBadRequest, "Invalid Config Map Key")
+		}
+
+		configMap, err := wfc.kubeclientset.CoreV1().ConfigMaps(items[0]).Get(items[2], metav1.GetOptions{})
+
+		if err != nil {
+			return 0, err
+		}
+
+		value, found := configMap.Data[items[3]]
+
+		if !found {
+			return 0, argoErr.New(argoErr.CodeBadRequest, "Invalid Sync configuration Key")
+		}
+		return strconv.Atoi(value)
+	}
+
+	wfc.concurrencyMgr = sync.NewLockManager(syncLimitConfig, func(key string) {
+		wfc.wfQueue.AddAfter(key, 0)
+	})
+
+	labelSelector := v1Label.NewSelector()
+	req, _ := v1Label.NewRequirement(common.LabelKeyPhase, selection.Equals, []string{string(wfv1.NodeRunning)})
+	if req != nil {
+		labelSelector = labelSelector.Add(*req)
+	}
+
+	listOpts := metav1.ListOptions{LabelSelector: labelSelector.String()}
+	wfList, err := wfc.wfclientset.ArgoprojV1alpha1().Workflows(wfc.namespace).List(listOpts)
+
+	wfc.concurrencyMgr.Initialize(wfList)
```

Comment:
What happens if `err` in line 241 is not nil? Why is there no `err` check before this?